db/upgrade.php

```diff
@@ -299,5 +299,24 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2023111500, 'cms');
     }
 
+    if ($oldversion < 2023112100) {
+        // Update CMS types settings to upgrade list to userlist.
+        $like = $DB->sql_like('datasources', ':list');
+        $records = $DB->get_records_select('cms_types', $like, ['list' => '%list%'], '', 'id,datasources');
```

Comment:
That's great, we need to update the `cms_types` but we also need to update the `cms` instances. E.g. this is the state that is broken right now
```
  id  | course |           name           | intro | introformat | typeid |                                                           customdata                                                           | usermodified | timecreated | timemodified 
------+--------+--------------------------+-------+-------------+--------+--------------------------------------------------------------------------------------------------------------------------------+--------------+-------------+--------------
  561 |    126 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"c6f58c8bd5b1cc3f6d242e8cc6de2ab523785be3","grade_items.cache_rev":13,"roles_course_role_cache_rev":1} |            2 |  1695960911 |   1697499322
  562 |    126 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"c6f58c8bd5b1cc3f6d242e8cc6de2ab523785be3","grade_items.cache_rev":12,"roles_course_role_cache_rev":1} |            2 |  1695960911 |   1697499322
  788 |    129 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"c948ca4aefd3d8e1bfda9c82753dab16177d522a","userlistinstanceids":[9,10]}                               |            2 |  1697546150 |   1697546172
  811 |    138 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"0898fb051be34dc259d593a5caa42a1315ee64ee","roles_course_role_cache_rev":1}                            |            2 |  1697691175 |   1697691181
  798 |    136 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"0898fb051be34dc259d593a5caa42a1315ee64ee"}                                                            |            2 |  1697548156 |   1697548156
  789 |    129 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"522fb0615bc0950553c2565b16af11a6923c932b","userlistinstanceids":[11,12]}                              |            2 |  1697546178 |   1697546195
  812 |    138 | Project Schedule Section |       |           0 |     17 | {"userlistinstancehash":"0898fb051be34dc259d593a5caa42a1315ee64ee","roles_course_role_cache_rev":1}                            |            2 |  1697691175 |   1697691181
 1069 |    145 | Project Schedule Section |       |           0 |     17 | {"listinstancehash":"0898fb051be34dc259d593a5caa42a1315ee64ee","rolesinstancerev":1}                                           |            2 |  1700453949 |   1700453954
 1070 |    145 | Project Schedule Section |       |           0 |     17 | {"listinstancehash":"0898fb051be34dc259d593a5caa42a1315ee64ee","rolesinstancerev":1}                                           |            2 |  1700453949 |   1700453954
(9 rows)
```
See how older records have `userlistinstancehash` while newer records have `listinstancehash` they both should just have `userlistinstancehash`